src/impl/collection_change_builder.hpp

```diff
@@ -51,6 +55,9 @@ class CollectionChangeBuilder : public CollectionChangeSet {
 
     // generic operations {
     CollectionChangeSet finalize() &&;
+
+    // Merge input change set into this one. Once merged the input builder should
+    // no longer be used.
```

Comment:
This is not true. It is perfectly fine to reuse the input builder, and we do so.

Reply:
Ups. I added this after running into the REALM_ASSET in `collection_notifier.add_changes()`, which does appear to consume the builder, but not sure why I added it here which is clearly a mistake. I'll move this comment.

Reply:
Now I remember why. It was because `add_changes` internally calls `merge` which was why I assumed this invariant also applied to `merge`, but I guess that is not the case?